Tests/CatalogItems/Write-RsFolderContent.Tests.ps1

```diff
@@ -0,0 +1,105 @@
+        $folderName = 'SutWriteRsFolderContentProxy' + [guid]::NewGuid()
+        New-RsFolder -Path / -FolderName $folderName
+        $folderPath = '/' + $folderName
+        $localReportPath =   (Get-Item -Path ".\" -Verbose).FullName  + '\Tests\CatalogItems\testResources'
+        $proxy = New-RsWebServiceProxy 
+        Write-RsFolderContent -Path $localReportPath -RsFolder $folderPath -Proxy $proxy
+        $uploadedReport = (Get-RsFolderContent -RsFolder $folderPath -Recurse ) | Where-Object TypeName -eq 'Report'
+        It "Should upload a local report in Report Server with Proxy Parameter" {
+            $uploadedReport.TypeName | Should Be 'Report'
+        }
+        # Removing folders used for testing
+        Remove-RsCatalogItem -RsFolder $folderPath
+    }
+
+    Context "Write-RsFolderContent with Proxy and ReportServerUri"{
+        $folderName = 'SutWriteRsFolderContentAll' + [guid]::NewGuid()
+        New-RsFolder -Path / -FolderName $folderName
+        $folderPath = '/' + $folderName
+        $localReportPath =   (Get-Item -Path ".\" -Verbose).FullName  + '\Tests\CatalogItems\testResources'
+        $proxy = New-RsWebServiceProxy 
+        $reportServerUri = 'http://localhost/reportserver'
+        Write-RsFolderContent -Path $localReportPath -RsFolder $folderPath -Proxy $proxy -ReportServerUri $reportServerUri
+        $uploadedReport = (Get-RsFolderContent -RsFolder $folderPath -Recurse ) | Where-Object TypeName -eq 'Report'
+        It "Should upload a local report in Report Server with ReportServerUri and Proxy Parameters" {
+            $uploadedReport.TypeName | Should Be 'Report'
+        }
+        # Removing folders used for testing
+        Remove-RsCatalogItem -RsFolder $folderPath
+    }
+
+     Context "Write-RsFolderContent with Recurse Parameter"{
+        $folderName = 'SutWriteRsFolderContentRecurse' + [guid]::NewGuid()
+        New-RsFolder -Path / -FolderName $folderName
+        $folderPath = '/' + $folderName
+        $localReportPath =   (Get-Item -Path ".\" -Verbose).FullName  + '\Tests\CatalogItems\testResources'
+        Write-RsFolderContent -Path $localReportPath -RsFolder $folderPath -Recurse
+       It "Should upload a local subFolder with Recurse Parameter" {
+            $uploadedDataSet = (Get-RsFolderContent -RsFolder $folderPath -Recurse ) | Where-Object TypeName -eq 'Folder'
+            $uploadedDataSet.TypeName | Should Be 'Folder'
+        }
+
+       It "Should upload a report that is in a folder and a second report in a subfolder" {
+            $uploadedReports = (Get-RsFolderContent -RsFolder $folderPath -Recurse ) | Where-Object TypeName -eq 'Report'
+            $uploadedReports.Count | Should Be 2
+        }
+
+         It "Should upload a local RsDataSource with Recurse Parameter" {
+            $uploadedDataSource = (Get-RsFolderContent -RsFolder $folderPath -Recurse ) | Where-Object TypeName -eq 'DataSource'
+            $uploadedDataSource.TypeName | Should Be 'DataSource'
```

Comment:
>$uploadedDataSource.TypeName | Should Be 'DataSource' [](start = 12, length = 53)

I recommend you to check for the name of the data source instead of TypeName as $uploadedDataSource is supposed to be an item of type 'DataSource'